src/common/exception.cpp

```diff
@@ -213,10 +213,10 @@ SequenceException::SequenceException(const string &msg) : Exception(ExceptionTyp
 InterruptException::InterruptException() : Exception(ExceptionType::INTERRUPT, "Interrupted!") {
 }
 
-FatalException::FatalException(const string &msg) : Exception(ExceptionType::FATAL, msg) {
+FatalException::FatalException(ExceptionType type, const string &msg) : Exception(type, msg) {
 }
 
-InternalException::InternalException(const string &msg) : Exception(ExceptionType::INTERNAL, msg) {
+InternalException::InternalException(const string &msg) : FatalException(ExceptionType::INTERNAL, msg) {
```

Comment:
I don't really understand the reason for this change, why is this needed?

Reply:
If I'm not mistaken this came out of a discussion I had with Mytherin, I think the general idea is that if an Internal Exception is triggered, that's similar to an assertion. At which point we cant guarantee that the database is in a valid state anymore.